packages/client/lib/rpc/modules/engine.ts

```diff
@@ -429,6 +457,18 @@ export class Engine {
         return { payloadStatus, payloadId: null }
       }
     }
+    const validTerminalBlock = await validateTerminalBlock(headBlock, this.chain)
```

Comment:
again this check should be done if headblock is a "pre-merge" block, check will fail on a post merge block